fiftyone/core/metadata.py

```diff
@@ -382,28 +389,79 @@ def compute_metadata(
             default value ``fiftyone.config.show_progress_bars`` (None), or a
             progress callback function to invoke instead
     """
-    num_workers = fou.recommend_thread_pool_workers(num_workers)
+    logger.info("Computing metadata...")
 
     if sample_collection.media_type == fom.GROUP:
         sample_collection = sample_collection.select_group_slices(
             _allow_mixed=True
         )
 
-    if num_workers <= 1:
-        _compute_metadata(
-            sample_collection, overwrite=overwrite, progress=progress
-        )
-    else:
-        _compute_metadata_multi(
+    if not overwrite:
+        sample_collection = sample_collection.exists("metadata", False)
+
+    if (sample_count := len(sample_collection)) == 0:
+        logger.info("No samples to compute metadata for")
```

Comment:
I request that this `if` block be removed. By convention, we don't log "nothing to do" messages in the FO SDK. We just silently return because there was nothing to do, which is a perfectly valid and successful outcome of the method.